# Relativistic Mass

This concept suggests that an object's mass increases as its velocity approaches the speed of light. However, this idea is outdated in modern physics. Mass normally refers to *rest mass* or *proper mass*, which is constant.

But lets explore the concept of relativistic mass in more depth.

## Terminology

- *Proper*: means your subjective experience of time, length and acceleration (eg on the ship that's moving). *Proper time* is time for you on ship
- *Coordinate*: means the time and length in a background frame of reference you think of as stationary (eg on Earth). *Coordinate time* is time for someone on Earth watching the ship
- *Rest mass*: is the real physical mass of an object when it is at rest relative to you (aka *proper mass*). *Rest mass* is the mass of an object on the ship, as you measure it on the ship

## Does an object's mass increase as its speed approaches c?

### And is this effect what stops us accelerating to light speed?

The answers here are 'not really' and 'no'!

## Relativistic mass increase

As objects speed up, their relativistic mass increases by the Lorentz factor (which also describes time dilation and length contraction). The Lorentz factor is given by:

$$\gamma = \frac{1}{\sqrt{1 - \frac{v^2}{c^2}}}$$

where $v$ is the relative velocity of the object and $c$ is the speed of light.

Here we calculate the Lorentz factor for a range of velocities:


In [43]:
import mpmath as mp
import relativity_lib as rl
import prettytable as pt

# adjust these velocities to see the effects of relativity
velocities = [
    0,
    1000,
    1000000,
    100000000,
    200000000,
    299000000,
    rl.c_float * 0.999,
    rl.c_float * 0.999999,
]
rest_mass = mp.mpf(1)

rl.configure(100)
minute = mp.mpf(60)
table = pt.PrettyTable(
    [
        "Velocity (m/s)",
        "Lorentz",
        "Relativistic mass (kg)",
        "Time dilation (1 min)",
        "Length contraction (1 m)",
    ]
)
table.align = "r"  # right align everything


def add_row(v):
    global rest_mass, table, minute

    # work out the relativistic mass, time dilation and length contraction
    gamma = rl.lorentz_factor(v)
    relativistic_mass = rest_mass * gamma
    time = minute * gamma
    length = rl.one / gamma

    # format the results
    fmt_gamma = rl.format_mpf_significant(gamma, 2)
    fmt_relativistic_mass = rl.format_mpf_significant(relativistic_mass, 1)
    fmt_time = rl.format_mpf_significant(time, 2) + " s"
    fmt_length = rl.format_mpf_significant(length * 100, 5) + " cm"

    # add the results to the table
    table.add_row([v, fmt_gamma, fmt_relativistic_mass, fmt_time, fmt_length])


# run the calculations
for v in velocities:
    add_row(v)

# print the table
print(table)

+------------------+-----------------+------------------------+-----------------------+--------------------------+
|   Velocity (m/s) |         Lorentz | Relativistic mass (kg) | Time dilation (1 min) | Length contraction (1 m) |
+------------------+-----------------+------------------------+-----------------------+--------------------------+
|                0 |             1.0 |                    1.0 |                60.0 s |                 100.0 cm |
|             1000 | 1.0000000000055 |         1.000000000005 |      60.00000000033 s |              99.99999 cm |
|          1000000 |       1.0000055 |               1.000005 |            60.00033 s |              99.99944 cm |
|        100000000 |           1.060 |                   1.06 |               63.64 s |              94.27274 cm |
|        200000000 |            1.34 |                    1.3 |               80.54 s |              74.49429 cm |
|        299000000 |           13.76 |                   13.7 |              825

## However..

Note that this increase is **frame dependent**. A 50kg object travelling at 0.999c will have the following relativistic mass when it hits the Earth:

In [44]:
# Work out relativistic mass of 50kg going at 0.999c
rest_mass = mp.mpf(50)

earth_velocity = 0.0
ship_velocity = 0.999


def calculate_relative_mass(vel1, vel2):
    global rest_mass
    relative_velocity_c = abs(vel2 - vel1)
    gamma = rl.lorentz_factor(rl.c * relative_velocity_c)
    relativistic_mass = rest_mass * gamma

    print(f"Rest mass: {rest_mass} kg")
    print(f"First velocity: {vel1}c, second velocity: {vel2}c")
    print(f"Relative velocity: {relative_velocity_c}c")
    print(f"Lorentz factor: {rl.format_mpf(gamma, 2)}")
    print(f"Relativistic mass: {rl.format_mpf(relativistic_mass, 2)} kg")


calculate_relative_mass(earth_velocity, ship_velocity)

Rest mass: 50.0 kg
First velocity: 0.0c, second velocity: 0.999c
Relative velocity: 0.999c
Lorentz factor: 22.36
Relativistic mass: 1,118.31 kg


But that's because the **relative velocity** between the object and Earth is 0.999c.

In the object's own frame (like on a spaceship moving with it), there is no relative motion. Its Lorentz factor is 1 and its relativistic mass is just:

In [45]:
# save frame of reference, whats the relativistic mass here?
object_velocity = 0.999
ship_velocity = 0.999

calculate_relative_mass(object_velocity, ship_velocity)

Rest mass: 50.0 kg
First velocity: 0.999c, second velocity: 0.999c
Relative velocity: 0.0c
Lorentz factor: 1.00
Relativistic mass: 50.00 kg


The relativistic mass is the same as the rest mass!

Lets consider a third frame of reference, this time an object travelling at 0.5c which the 0.999c object collides with:

In [46]:
object_velocity = 0.999
mid_velocity = 0.5

calculate_relative_mass(object_velocity, mid_velocity)

Rest mass: 50.0 kg
First velocity: 0.999c, second velocity: 0.5c
Relative velocity: 0.499c
Lorentz factor: 1.15
Relativistic mass: 57.69 kg


The relativistic mass is different again! Depending on your point of view, it seems to be:

- 50 kg
- 57.69 kg
- 1,118.31 kg

## But how can it have three relativistic masses at once?

This is the key concept. In 10 different frames of reference, an object will have 10 different relativistic masses **simultaneously**. None of them is more real that the others.

This is why physicists say "relativistic mass" is misleading.

### So what do you get when you weigh the object travelling at 0.999c (measure its mass)?

The scale is in the same frame as the object, so the Lorentz factor is 1 and the measured mass is just the rest mass. It weighs 1kg regardless of its velocity towards the Earth, or any other object.

## Lorentz factors are always between 2 specific frames

For example, the Earth and the ship.

Whenever you use a Lorentz factor you know you are calculating something frame dependent. This includes:

- Time dilation
- Length contraction
- Relativistic energy and mass

All these effects have multiple values simultaneously, according to the frames you choose to consider. Time may slow down by 10% in one frame, but its slowed by 50% and 99% in other frames you could choose.

# Particle accelerators and light sails

If you are accelerating something in a different reference frame, like a particle accelerator moving protons, or a Earth-based laser shining on a ship with a light sail, relativistic mass does indeed come into play. At the object speeds up (relative to the accelerator) the Lorentz factor increases, and it gets harder to accelerate.

It would require infinite energy or time to accelerate it to c.

# Ship's engines

This is very different.

The engines and fuel (and you the captain) are all moving together. Your velocity relative to the engines always remains 0. It doesn't matter that you and the ship may be travelling at 0.999c from a particular frame, you can't discover that while on board (unless you look to see how fast the Earth is approaching, and then it could be the Earth thats moving and not you).

## No absolute motion, it's all relative

When not accelerating, all inertial frames are equivalent. There is no test you can perform to find your "true" velocity, you are essentially stationary.

This is the case whether your speed is 0 or 0.999c relative to Earth.

Everything on board ship remains exactly the same. Its the outside universe (in a different frame) that changes.

# So why can't we reach light speed? If it's not relativistic mass that stops us?

Remember that all inertial frames are equivalent. You accelerate to 0.999c (relative to the Earth) and measure light speed...

### ..and it's just as fast as always!

You have made zero progress towards matching it, and you never will! This is why c is unique, its the same in all frames regardless of relative motion.

299,792,458 metres per second

## In your own frame you can accelerate forever

So in your own frame, there is no universal speed limit. You can literally accelerate forever and cross the universe in a day (if the g-force didn't kill you).

See the other notebook about a tour of the universe that takes just a few decades. You can travel over 100 billion light years in a human lifetime at 1g acceleration.

## In coordinate time

In a coordinate frame (eg Earth), the ship's acceleration would seem to slow as your speed approaches c. Time on board ship would also slow to a standstill as your velocity increases:

In [47]:
# In coordinate time, what distance is traveled in each year under proper constant acceleration?

accel = rl.g  # 9.81 m/s^2
minute = mp.mpf(60)  # 1 minute in seconds

# Used in the loop
previous_total = mp.mpf(0)
previous_year = mp.mpf(0)

years = pt.PrettyTable(
    [
        "Coordinate time (year number)",
        "Additional km",
        "Lorentz",
        "1 minute ship-time",
        "This year km",
    ]
)
years.align = "r"  # right align everything

for year in range(1, 40):
    # work out total distance in coord time for journey of year duration
    sec = mp.mpf(year) * rl.seconds_per_year
    total_distance = rl.relativistic_distance_coord(accel, sec)

    # work out the velocity (coord time) and lorentz factor
    velocity = rl.relativistic_velocity_coord(accel, sec)
    gamma = rl.lorentz_factor(velocity)

    # time dilation factor
    time_dilation = minute * gamma

    # distance traveled in this year
    this_year = total_distance - previous_total

    # the increase this year
    this_year_increase = this_year - previous_year

    # update the previous values
    previous_total = total_distance
    previous_year = this_year

    # format the results and add to table
    fmt_this_year = rl.format_mpf(this_year_increase / 1000, 1)
    fmt_this_year2 = rl.format_mpf(this_year / 1000, 1)
    fmt_gamma = rl.format_mpf(gamma, 1)
    fmt_time = rl.format_mpf(time_dilation, 2) + " s"
    years.add_row([year, fmt_this_year, fmt_gamma, fmt_time, fmt_this_year2])

print(
    "In coordinate time under constant proper acceleration, the distance traveled in each year is:"
)
print()
print(years)

In coordinate time under constant proper acceleration, the distance traveled in each year is:

+-------------------------------+---------------------+---------+--------------------+---------------------+
| Coordinate time (year number) |       Additional km | Lorentz | 1 minute ship-time |        This year km |
+-------------------------------+---------------------+---------+--------------------+---------------------+
|                             1 | 4,007,115,677,439.6 |     1.4 |            86.23 s | 4,007,115,677,439.6 |
|                             2 | 3,845,154,131,734.4 |     2.2 |           137.64 s | 7,852,269,809,174.0 |
|                             3 |   948,774,306,887.1 |     3.2 |           195.26 s | 8,801,044,116,061.2 |
|                             4 |   310,638,564,220.0 |     4.2 |           254.91 s | 9,111,682,680,281.2 |
|                             5 |   134,730,061,290.0 |     5.2 |           315.44 s | 9,246,412,741,571.2 |
|                             6 |

Notice how the extra distance travelled each year shrinks as the Lorentz factor increases even though the proper acceleration is constant.

## However in proper time

But doing the same calculations in proper time (each year as it passes on board ship) gives a different result:

In [48]:
# In proper time, what distance is traveled in each year under proper constant acceleration?

accel = rl.g  # 9.81 m/s^2
minute = mp.mpf(60)  # 1 minute in seconds

# Used in the loop
previous_total = mp.mpf(0)
previous_year = mp.mpf(0)

years = pt.PrettyTable(["Proper time (year number)", "Additional km", "This year km"])
years.align = "r"  # right align everything

for year in range(1, 40):
    # work out total distance in coord time for journey of year duration
    sec = mp.mpf(year) * rl.seconds_per_year
    total_distance = rl.relativistic_distance(accel, sec)

    # work out the velocity (coord time) and lorentz factor
    # velocity = rl.relativistic_velocity(accel, sec)
    # gamma = rl.lorentz_factor(velocity)

    # time dilation factor
    # time_dilation = minute * gamma

    # distance traveled in this year
    this_year = total_distance - previous_total

    # the increase this year
    this_year_increase = this_year - previous_year

    # update the previous values
    previous_total = total_distance
    previous_year = this_year

    # format the results and add to table
    fmt_this_year = rl.format_mpf(this_year_increase / 1000, 1)
    fmt_this_year2 = rl.format_mpf(this_year / 1000, 1)
    # fmt_gamma = rl.format_mpf(gamma, 1)
    # fmt_time = rl.format_mpf(time_dilation, 2) + " s"
    years.add_row([year, fmt_this_year, fmt_this_year2])

print(
    "However in proper time under constant proper acceleration, the distance traveled in each year is:"
)
print()
print(years)

However in proper time under constant proper acceleration, the distance traveled in each year is:

+---------------------------+-------------------------------------------+-------------------------------------------+
| Proper time (year number) |                             Additional km |                              This year km |
+---------------------------+-------------------------------------------+-------------------------------------------+
|                         1 |                       5,332,469,445,011.0 |                       5,332,469,445,011.0 |
|                         2 |                      16,870,285,098,259.3 |                      22,202,754,543,270.3 |
|                         3 |                      42,707,428,684,034.0 |                      64,910,183,227,304.3 |
|                         4 |                     118,242,818,716,579.8 |                     183,153,001,943,884.2 |
|                         5 |                     331,376,301,779,117.7 |  

# Conclusion

While we still have fuel, we can accelerate forever in our own frame. The speed of light doesn't stop us, we simply never get close.

In *proper* years (years on ship) we move faster each year as time dilation occurs (in the 39th years at 1g we travel $900,191,726,353,464,633,490,928,654,862 km$).

However in *coordinate* years (years on Earth) we appear to slow down as we approach c even though the *proper* acceleration is constant. In the 39th year we only travel $9,457,736,606,590.1 km$.

The 'relativistic mass' of the ship doesn't increase as far as the ship is concerned, and this isn't what stops us reaching light speed. We can't reach light speed because its equally fast in all frames, and we never get close to it.